docs/instrument.md

```diff
@@ -0,0 +1,49 @@
+
+## Producing instrumented source
```

Comment:
We can up all the headers in this file now that it's alone :) 

There's also an extra newline here